tests/unit/test_main.py

```diff
@@ -564,6 +564,22 @@ def test_parse_vlan(self):
         vlan, msg = self.napp.parse_vlan("1:9999")
         assert vlan is None
         assert "Invalid vlan" in msg
+        # case 6: range - valid (start == end)
+        vlan, msg = self.napp.parse_vlan("100:100")
+        assert vlan == [[100, 100]]
+        assert msg is None
+        # case 7: range - invalid (start > end)
+        vlan, msg = self.napp.parse_vlan("100:99")
+        assert vlan is None
+        assert "Invalid vlan range" in msg
+        # case 8: range - invalid format (non-numeric)
+        vlan, msg = self.napp.parse_vlan("a:b")
+        assert vlan is None
+        assert "Invalid vlan range" in msg
+        # case 9: range - invalid format (missing colon)
+        vlan, msg = self.napp.parse_vlan("100-200")
+        assert vlan is None
+        assert "Invalid vlan range" in msg
```

Comment:
Looks like `"100.2:100.1"` would also be accepted and mapped into `[100, 100]`. Is a cast from string float to int be expected to be cast indeed or should it be a validation error too?

Reply:
Hi @viniarck, thank you very much for your comments.

You are right: I added the test both for the case of strings with float and for the scenario where sdx_vlan is an int. After the changes, all tests pass.